### PR TITLE
Add documentation about manipulating console output

### PR DESCRIPTION
**Pull request by pierredup**
Add documentation about manipulating console output (symfony/symfony#24363)

### CODE REVIEW

File: console/manipulating_output.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
+How To Manipulate Console Output
+================================
+
+The console comes with a powerful class to print information to the terminal. This
+information can be manipulated by clearing or overwriting the displayed content.
+
+In order to manipulate the content, you need to create a new output section. An output section is
+a part in the terminal where information will be displayed from the console.
+
+A section can be manipulated individually, and multiple sections can appended to the output.
```
Comment (fabpot) [inline console/manipulating_output.rst]: `can be`?

File: console/manipulating_output.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    class MyCommand extends Command
+    {
+        protected function execute(InputInterface $input, OutputInterface $output)
+        {
+            $section = $output->section();
+        }
+    }
+}
+
+This will return an instance of of the :class:`Symfony\\Component\\Console\\Output\\ConsoleSectionOutput`
```
Comment (fabpot) [inline console/manipulating_output.rst]: double `of`

File: console/manipulating_output.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    You can create multiple sections by calling the
+    :method:`Symfony\\Component\\Console\\Output\\ConsoleOutput::section` method multiple times.
+    This will append a new section after the previous one.
+
+.. caution::
+
+    Displaying information in a section will always append a new line to the output.
+
+
```
Comment (fabpot) [inline console/manipulating_output.rst]: double blank line

File: console/manipulating_output.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+When displaying information in the console, you can overwrite the output by using the
+:method:`Symfony\\Component\\Console\\Output\\ConsoleSectionOutput::overwrite` method::
+
+    $section->writeln('Hello');
+    $section->overwrite('World!');
+
+The only information displayed in the terminal will be ``World!`` as the first part will
+be overwritten.
+
+Clearing s Section
```
Comment (fabpot) [inline console/manipulating_output.rst]: `a`?

File: console/manipulating_output.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+------------------
+
+You can clear all the content in a section by using the
+:method:`Symfony\\Component\\Console\\Output\\ConsoleSectionOutput::overwrite` method.
```
Comment (fabpot) [inline console/manipulating_output.rst]: `clear`?

**Review by fabpot — CHANGES_REQUESTED** [review]

**Review by ndench — APPROVED** [review]
Looks good to me. Thanks for the contribution.

**Review by ndench — APPROVED** [review]
Very useful. Thanks!

**Comment by javiereguiluz** [comment]
@pierredup thanks a lot for contributing these docs! (and for implementing the feature too ... it's amazing!) I like the contents, but I'd prefer another organization, so I'm going to merge this PR "as is" and then I'll create a different PR to reorganize contents. Cheers!

**Pull request closed without merging**